src/coreclr/vm/codeman.cpp

```diff
@@ -396,8 +396,8 @@ void UnwindInfoTable::AddToUnwindInfoTable(UnwindInfoTable** unwindInfoPtr, PT_R
     _ASSERTE(pRS != NULL);
     if (pRS != NULL)
     {
-        _ASSERTE(pRS->_pjit->GetCodeType() == (miManaged | miIL));
-        if (pRS->_pjit->GetCodeType() == (miManaged | miIL))
+        _ASSERTE(pRS->_pjit->GetCodeType() == (miManaged | miIL) || pRS->_pjit->GetCodeType() == (miManaged | miIL | miOPTIL));
+        if (pRS->_pjit->GetCodeType() == (miManaged | miIL)) // Do this only for Jitted code, not for interpreted code
```

Comment:
Why don't we want the unwind info to get cleaned up? I only see this method get called from LCGMethodResolver::TryDestroyCodeHeapMemory() and my prior understanding is that we expect LCG methods to be cleaned up when we are done with them.

Separately, I wouldn't expect a method called 'Unpublish' to sometimes decide not to Unpublish. If this method gets to decide how to react perhaps the method name should be what its reacting to? For example UnwindInfoTable::HandleMethodCleanup()? Or the caller could have the decision making logic?

Reply:
I am also not sure I understand why we need to have the cleanup like this, the interpreted code doesn't have any unwind info at all.

Reply:
Ah, that makes more sense. If the comment said something like "There is no unwind info to unpublish" I think it would be more easily understood.